tools/tiled_ensemble/test_ensemble.py

```diff
@@ -0,0 +1,122 @@
+# SPDX-License-Identifier: Apache-2.0
+
+import logging
+import warnings
+from argparse import ArgumentParser, Namespace
+from itertools import product
+from pathlib import Path
+
+from omegaconf import OmegaConf
+from pytorch_lightning import Trainer, seed_everything
+from tools.tiled_ensemble import (
+    EnsembleTiler,
+    get_ensemble_callbacks,
+    get_ensemble_datamodule,
+    get_prediction_storage,
+    log_metrics,
+    post_process,
+    prepare_ensemble_configurable_parameters,
+)
+
+from anomalib.config import get_configurable_parameters
+from anomalib.models import get_model
+from anomalib.utils.loggers import configure_logger, get_experiment_logger
+
+logger = logging.getLogger("anomalib")
+
+
+def get_parser() -> ArgumentParser:
+    """Get parser.
+
+    Returns:
+        ArgumentParser: The parser object.
+    """
+    parser = ArgumentParser()
+    parser.add_argument("--model", type=str, default="padim", help="Name of the algorithm to train/test")
+    parser.add_argument("--config", type=str, required=False, help="Path to a model config file")
+    parser.add_argument("--ens_config", type=str, required=True, help="Path to an ensemble configuration file")
+    parser.add_argument("--weight_folder", type=str, required=True, help="Path to directory with saved weight files")
+    parser.add_argument("--log-level", type=str, default="INFO", help="<DEBUG, INFO, WARNING, ERROR>")
+
+    return parser
+
+
+def test(args: Namespace):
+    """Test a tiled ensemble.
+
+    Images are split to tiles and separate model is used to predict for each tile location.
+
+    Args:
+        args (Namespace): The arguments from the command line.
+    """
+
+    configure_logger(level=args.log_level)
+
+    if args.log_level == "ERROR":
+        warnings.filterwarnings("ignore")
+
+    config = get_configurable_parameters(model_name=args.model, config_path=args.config)
+    # update and prepare config for ensemble
+    config = prepare_ensemble_configurable_parameters(ens_config_path=args.ens_config, config=config)
+
+    experiment_logger = get_experiment_logger(config)
+
+    # instantiate tiler used for splitting images into tiles
+    tiler = EnsembleTiler(
+        tile_size=config.ensemble.tiling.tile_size,
+        stride=config.ensemble.tiling.stride,
+        image_size=config.dataset.image_size,
+    )
+    # prepare datamodule with tiling mechanism
+    datamodule = get_ensemble_datamodule(config, tiler)
+    # prepare storage objects that handle storage of tiled predictions
+    ensemble_predictions, _ = get_prediction_storage(config)
+
+    logger.info(
+        "Tiled ensemble testing started. Separate models will be evaluated for %d tile locations.",
+        tiler.num_patches_h * tiler.num_patches_w,
+    )
+    # go over all tile positions and test
+    for tile_index in product(range(tiler.num_patches_h), range(tiler.num_patches_w)):
+        logger.info("Start of procedure for tile %s", tile_index)
```

Comment:
If we have `tiler.num_tiles`, we could potentially use it here as well, like:

```python
logger.info("Start of procedure for tile %s of %s", tile_index, tiler.num_tiles)
```

Reply:
done